dns/pfSense-pkg-bind/files/usr/local/pkg/bind.inc

```diff
@@ -586,9 +586,24 @@ EOD;
 							$zone_conf .= "\n\n;\n;custom zone records\n;\n".base64_decode($zone['customzonerecords'])."\n";
 						}
 
+						// Freeze dynamic zones to prevent journal corruption
+						$zone_is_dynamic = $zone['enable_updatepolicy'] == "on" || $zoneallowupdate != "none";
+						$rndc_conf_path = BIND_LOCALBASE . "/etc/rndc.conf";
+						$rndc = "/usr/local/sbin/rndc -q -c $rndc_conf_path";
+						if ($zone_is_dynamic) {
+							exec("$rndc freeze $zonename IN $zoneview");
```

Comment:
Since this is a shell exec using user input (zone name and view), each of these variable uses should be wrapped in `escapeshellarg()`, for example:

    exec("$rndc freeze " . escapeshellarg($zonename) . " IN " . escapeshellarg($zoneview));